x/mongo/driver/topology/topology_test.go

```diff
@@ -230,6 +235,30 @@ func TestServerSelection(t *testing.T) {
 			t.Fatalf("did not receive error from server selection")
 		}
 	})
+	t.Run("Context Deadline Error", func(t *testing.T) {
+		desc := description.Topology{}
+		topo, err := New()
+		noerr(t, err)
+		subCh := make(chan description.Topology, 1)
+		subCh <- desc
+		resp := make(chan error)
+		timeout := make(chan time.Time)
+		go func() {
+			state := newServerSelectionState(selectContextError, timeout)
+			ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
```

Comment:
nit: 1 second is a relatively long test execution time. Can this be `50*time.Millisecond` or does the time matter?

Reply:
done. Changed to 50*time.Millisecond